### PR TITLE
Initial dataset site implementation

### PR DESCRIPTION
**Pull request by ylt**

### CODE REVIEW

File: lib/openactive/dataset.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# require "openactive/dataset/feed_configuration"
+require "openactive/dataset/feed_type"
+require "openactive/dataset/meta"
+require "openactive/dataset/settings"
+require "openactive/dataset/template_renderer"
+
+module OpenActive
+  module Dataset
+    class Error < StandardError; end
+    # Your code goes here...
```
Comment (henryaddison) [inline lib/openactive/dataset.rb]: You can remove this line

File: .ruby-version [hunk 2]
```diff
@@ -0,0 +1 @@
+2.6.3
```
Comment (henryaddison) [inline .ruby-version]: Don't include a .ruby-version file. As a library even contributors shouldn't be tied to a ruby version

File: openactive-dataset.gemspec [hunk 3]
```diff
@@ -0,0 +1,34 @@
+lib = File.expand_path('lib', __dir__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require "openactive/dataset/version"
+
+Gem::Specification.new do |spec|
+  spec.name          = "openactive-dataset"
+  spec.version       = OpenActive::Dataset::VERSION
+  spec.authors       = ["Joe Carter"]
+  spec.email         = ["joe@simpleweb.co.uk"]
```
Comment (henryaddison) [inline openactive-dataset.gemspec]: Copy from npm modules

File: openactive-dataset.gemspec [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  spec.summary       = 'Write a short summary, because RubyGems requires one.'
```
Comment (henryaddison) [inline openactive-dataset.gemspec:11]: Update this summary - may want to ask Nick or Tim or copy from PHP/.NET

File: lib/openactive/dataset/settings.rb [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+      attr_accessor :platform_url
+      attr_accessor :platform_software_version
+
+      attr_accessor :data_feed_types
+
+      def data_feed_descriptions
+        data_feed_types.map do |description|
+          next description.value if description.respond_to?(:value)
+
+          description
```
Comment (henryaddison) [inline lib/openactive/dataset/settings.rb]: `description.respond_to?(:value) ? description.value : description`

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,11 @@
+/.bundle/
```
Comment (henryaddison) [inline .gitignore:1]: Ignore .ruby-version (see below) and Gemfile.lock too

**Review by henryaddison — COMMENTED** [review]
I'm made some comments inline, mainly tidying up ahead of a public release. Also make sure to include README (check the PHP version).

File: openactive-dataset.gemspec (hunk 4 above)
Comment (ylt) [reply openactive-dataset.gemspec:11]: @nickevansuk Small one for you. Can't see any existing package description for the .NET/PHP implementations.

File: lib/openactive/dataset.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+require "openactive"
+require "openactive/dataset/version"
+
+# require "openactive/dataset/feed_configuration"
```
Comment (henryaddison) [inline lib/openactive/dataset.rb]: Remove this line

File: openactive-dataset.gemspec [hunk 8]
```diff
@@ -0,0 +1,34 @@
+lib = File.expand_path('lib', __dir__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require "openactive/dataset/version"
+
+Gem::Specification.new do |spec|
+  spec.name          = "openactive-dataset"
+  spec.version       = OpenActive::Dataset::VERSION
+  spec.authors       = ["Joe Carter"]
```
Comment (henryaddison) [inline openactive-dataset.gemspec]: Based on their npm packages, include "OpenActive Community" in authors

File: openactive-dataset.gemspec (hunk 3 above)
Comment (henryaddison) [inline openactive-dataset.gemspec]: Based on their npm packages, include hello@openactive.io in email

File: Gemfile [hunk 9]
```diff
@@ -0,0 +1,10 @@
+source "https://rubygems.org"
+
+git_source(:github) { |repo_name| "https://github.com/#{repo_name}" }
+
+# Uncomment one of the below to aid with development:
+# gem "openactive", git: "https://github.com/openactive/models-ruby.git"
+# gem "openactive", path: "#{__dir__}/../models-ruby"
```
Comment (henryaddison) [inline Gemfile]: Remove this from Gemfile. If worried about helping contributors add something to Contributing section of README.

**Review by henryaddison — APPROVED** [review]
Happy for this PR to be merged. We will get a short summary off Nick later.

**Pull request merged**